### PR TITLE
Sync with Cirq master branch

### PR DESCRIPTION
**Pull request by kevinsung**
- Use cirq.decompose protocol instead of CompositeGate.
- Replace RotZGate with ZPowGate or Rz. In some cases replace Rot11Gate with CZPowGate.
- Add ofc.rot11 to replace former cirq.Rot11Gate.

### CODE REVIEW

File: openfermioncirq/primitives/bogoliubov_transform.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def _slater_basis_change(qubits: Sequence[cirq.QubitId],
         circuit_description = list(reversed(decomposition))
         # The initial state is not a computational basis state so the
         # phases left on the diagonal in the decomposition matter
-        yield (cirq.RotZGate(rads=numpy.angle(diagonal[j])).on(qubits[j])
+        yield (cirq.Rz(rads=numpy.angle(diagonal[j])).on(qubits[j])
```
Comment (Strilanc) [inline openfermioncirq/primitives/bogoliubov_transform.py]: `rads=` not required for `Rz`

File: openfermioncirq/gates/four_qubit_gates.py [hunk 2]
```diff
@@ -311,12 +311,12 @@ def default_decompose(self, qubits):
             ]))
 
         controlled_Zs = list(cirq.flatten_op_tree([
-            cirq.Rot11Gate(half_turns=exponents[0])(b, c),
+            cirq.CZPowGate(exponent=exponents[0])(b, c),
```
Comment (Strilanc) [inline openfermioncirq/gates/four_qubit_gates.py]: `cirq.CZ(b, c)**exponents[0]`

File: openfermioncirq/trotter/algorithms/split_operator_test.py [hunk 3]
```diff
@@ -215,13 +212,13 @@ def test_split_operator_trotter_step_asymmetric():
 │       │           │           │
 Z^0.0   Z^0.0       #2──────────YXXY
 │       │           │           │
-│       #2──────────YXXY^0.333  Z
+│       #2──────────YXXY^0.333  Z^-1
```
Comment (Strilanc) [inline openfermioncirq/trotter/algorithms/split_operator_test.py]: Report this as a bug. It shouldn't be adding `-1` here.

Comment (kevinsung) [reply openfermioncirq/trotter/algorithms/split_operator_test.py]: Opened https://github.com/quantumlib/Cirq/issues/1080

**Pull request merged**